### PR TITLE
add netcdf-cxx4 recipe

### PR DESCRIPTION
**Pull request by kmuehlbauer**
This adds the v4.2.1 (latest) version of the netcdf c++ libraries.

Pre-tested locally on linux-64 without flaws

I gladly add volunteers for the maintainer section.

### CODE REVIEW

File: recipes/netcdf-cxx4/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  name: netcdf-cxx4
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/Unidata/netcdf-cxx4
+  git_tag: v{{ version }}
+
+build:
+  number: 0
+  skip: True  # [win]
```
Comment (ocefpaf) [inline recipes/netcdf-cxx4/meta.yaml]: As soon as the feedstock is created let's open an issue to tackle the Windows build.

Ping @groutr who might have an idea about this.

File: recipes/netcdf-cxx4/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - ncxx4-config --all
+
+about:
+  home: https://github.com/Unidata/netcdf-cxx4
+  license: http://www.unidata.ucar.edu/software/netcdf/copyright.html
+  summary: Unidata NetCDF C++ Library
+
+extra:
+  recipe-maintainers:
+    - kmuehlbauer
```
Comment (ocefpaf) [inline recipes/netcdf-cxx4/meta.yaml]: You can add me (and maybe @groutr?).

File: recipes/netcdf-cxx4/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - libnetcdf
+  run:
+    - libnetcdf
```
Comment (ocefpaf) [inline recipes/netcdf-cxx4/meta.yaml]: We have #213 pending. (My bad.)

Can you take a look? Do you want to be a maintainer of that one too?

**Comment by ocefpaf** [comment]
Awesome addition @kmuehlbauer! Thanks!!

File: recipes/netcdf-cxx4/meta.yaml (hunk 2 above)
Comment (kmuehlbauer) [inline recipes/netcdf-cxx4/meta.yaml]: Will do when the feedstock is available.

File: recipes/netcdf-cxx4/meta.yaml (hunk 3 above)
Comment (kmuehlbauer) [inline recipes/netcdf-cxx4/meta.yaml]: If I can help in any way, yes. I'll have closer look in a few days after finishing some other pending stuff.

**Pull request merged**